lib/lte_link_control/lte_lc.c

```diff
@@ -68,6 +68,7 @@ static char rai_param[2] = CONFIG_LTE_RAI_REQ_VALUE;
 /* Requested NCELLMEAS params */
 static struct lte_lc_ncellmeas_params ncellmeas_params;
 static bool ncellmeas_ongoing;
+K_MUTEX_DEFINE(ncellmeas_ongoing_mutex);
```

Comment:
I guess it depends on what happens in the event handlers. Maybe a higher priority thread is waiting and gets to run and issues a new ncell request before the flag is cleared. Would be interesting to understand what is actually going on

Reply:
Commit description says what I know. When I'm printing ncellmeas_ongoing, it looks like it should be cleared in the event handler before another thread issues new `lte_lc_neighbor_cell_measurement()`. I cannot figure out why it behaves the way it does even with volatile and atomic_t.

Reply:
Mutex doesn't work nicely because you cannot ask for it's state. I've now used semaphore and seems to work nicely. If I set K_NO_WAIT to semaphore timeout, it'll fail as earlier but works nicely with 1s waiting time. Few milliseconds would probably be enough but 1s should be fine.

Reply:
I need to see if this implementation without 2nd ncellmeas triggered from a separate work queue is something I can use in unit tests as they seem to fail now.

Reply:
Unit tests are now also updated and the issues were related to the fact I forgot to give the semaphore in ncellmeas cancel.